sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java

```diff
@@ -595,23 +602,88 @@ private static String pageRangeToString(PageRange pageRange) {
         BlobRequestConditions requestConditions) {
         try {
             return withContext(context ->
-                getPageRangesDiffWithResponse(blobRange, prevSnapshot, requestConditions, context));
+                getPageRangesDiffWithResponse(blobRange, prevSnapshot, null, requestConditions, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * This API only works for managed disk accounts.
+     * <p>Gets the collection of page ranges that differ between a specified snapshot and this page blob. For more
+     * information, see the <a href="https://docs.microsoft.com/rest/api/storageservices/get-page-ranges">Azure
+     * Docs</a>.</p>
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.specialized.PageBlobAsyncClient.getManagedDiskRangesDiff#BlobRange-String}
+     *
+     * @param blobRange {@link BlobRange}
+     * @param prevSnapshotUrl Specifies the URL of a previous snapshot of the target blob. Specifies that the
+     * response will contain only pages that were changed between target blob and previous snapshot. Changed pages
+     * include both updated and cleared pages. The target blob may be a snapshot, as long as the snapshot specified by
+     * prevsnapshot is the older of the two.
+     *
+     * @return A reactive response emitting all the different page ranges.
+     */
+    public Mono<PageList> getManagedDiskRangesDiff(BlobRange blobRange, String prevSnapshotUrl) {
+        try {
+            return getManagedDiskRangesDiffWithResponse(blobRange, prevSnapshotUrl, null).flatMap(FluxUtil::toMono);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * This API only works for managed disk accounts.
```

Comment:
I'd prefer to see a more descriptive first sentence for some of these new JavaDocs. The first sentence gets extra prominence in the generated JavaDoc, so it is important to give a useful first sentence.

Reply:
This is by design.  We want the fact that this API only works with managed disk accounts to be prominently featured in the generated JavaDoc.  The vast majority of users shouldn't attempt to use this method.

Reply:
Even still, when you're only given the first-sentence description at the top of JavaDoc, I don't want to see one or potentially many methods with a description that only says this. I would rather it say something like "This API, which only works for managed disk accounts, allows the user to ....".